src/XRP/model/xrp-path-element.ts

```diff
@@ -19,8 +20,28 @@ export default class XrpPathElement {
   public static from(pathElement: PathElement): XrpPathElement {
     const account = pathElement.getAccount()?.getAddress()
     const currency = pathElement.getCurrency()
+    if (account && currency) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'PathElement protobuf should not contain both `account` and `currency` fields.',
+      )
+    }
+
     const xrpCurrency = currency && XrpCurrency.from(currency)
```

Comment:
can we move this closer to where it's being used? i expected it to be used in the next check